warrior/Actions/SeleniumActions/browser_actions.py

```diff
@@ -186,13 +208,23 @@ def browser_launch(self, system_name, browser_name="all", type="firefox",
             browser_details = arguments
 
         for browser in browser_list:
-            arguments = Utils.data_Utils.get_default_ecf_and_et(arguments, self.datafile, browser)
+            arguments = Utils.data_Utils.get_default_ecf_and_et(arguments,
+                                                                self.datafile,
+                                                                browser)
             if browser_details == {}:
                 browser_details = selenium_Utils.\
                     get_browser_details(browser, datafile=self.datafile, **arguments)
             if browser_details is not None:
-                browser_inst = self.browser_object.open_browser(
-                    browser_details["type"], webdriver_remote_url)
+                if type == "firefox":
+                    ff_profile = self.browser_object.\
+                        set_firefoxprofile(proxy_ip, proxy_port)
+                if binary != "" and gecko_path != "":
```

Comment:
gecko_path and binary value may be None

Reply:
These issues were found and addressed in the PR : https://github.com/warriorframework/warriorframework/pull/317

Reply:
The PR mentioned is looking for gecko_path and binary in system, but gecko_path and binary should be in browser, which is a child of system
so it isn't addressed correctly